js/main.js

```diff
@@ -1,382 +0,0 @@
-// Weather App
-// by Roger Mullins
-
-// Object Definitions
-
-let stateObj = {
-  city : "",
-  zipCode: "",
-  tempK: 0,
-  tempF: 0,
-  tempC: 0,
-  condition: "",
-  imageURL: ""
-};
-
-let lookupHistoryObj = {
-  city: "",
-  zipCode: ""
-};
-
-let localStorageObj = {
-  city: "",
-  zipCode: ""
-}
-
-// Global Variable
-
-let zipIn = "";   // Used to validate ZIP
-let tempF = 0;    // Used to update temperature background color
-
-function buildUI() {
-  // Creates HTML elements
-
-  function createElement(parentID, typeOfElement, bootstrapClasses, idString, innerText = "") {
-    let target = document.getElementById(parentID);
-    let newElement = document.createElement(typeOfElement);
-    newElement.classList.add(...bootstrapClasses);
-    newElement.setAttribute("id", idString);
-    if (innerText != "") {
-      newElement.innerText = innerText;
-    }
-    target.appendChild(newElement);
-  };
-
-  function createNavToggle() {
-    let target = document.getElementById('nav-bar-container');
-    let newButton = document.createElement('button');
-    newButton.classList.add('navbar-toggler');
-    newButton.setAttribute('type', 'button');
-    newButton.setAttribute('data-bs-toggle', 'collapse');
-    newButton.setAttribute('data-bs-target', '#nav-menu');
-    newButton.setAttribute('id', 'hamburger');
```

Comment:
A cool way to write a function for this would be:
```javascript
// pass an element and an object.
// Loop through the object and assign attributes and values
// based on key, value pairs.
const setMiscAttributes = (el, obj) => {
    for (const [key, value] of Object.entries(obj)) {
        el.setAttribute(key, value);
    }
}

setMiscAttributes(errorDiv, { role: 'alert', style: 'display: none; text-transform: capitalize;' });
```